shell/platform/linux/fl_json_method_codec.cc

```diff
@@ -0,0 +1,202 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_json_method_codec.h"
+
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_json_message_codec.h"
+
+#include <gmodule.h>
+
+struct _FlJsonMethodCodec {
+  FlMethodCodec parent_instance;
+
+  FlJsonMessageCodec* codec;
+};
+
+G_DEFINE_TYPE(FlJsonMethodCodec,
+              fl_json_method_codec,
+              fl_method_codec_get_type())
+
+static void fl_json_method_codec_dispose(GObject* object) {
+  FlJsonMethodCodec* self = FL_JSON_METHOD_CODEC(object);
+
+  g_clear_object(&self->codec);
+
+  G_OBJECT_CLASS(fl_json_method_codec_parent_class)->dispose(object);
+}
+
+// Implements FlMethodCodec::encode_method_call
+static GBytes* fl_json_method_codec_encode_method_call(FlMethodCodec* codec,
+                                                       const gchar* name,
+                                                       FlValue* args,
+                                                       GError** error) {
+  FlJsonMethodCodec* self = FL_JSON_METHOD_CODEC(codec);
+
+  g_autoptr(FlValue) message = fl_value_new_map();
+  fl_value_set_take(message, fl_value_new_string("method"),
```

Comment:
I'd prefer using file-level static constants for these so we don't have multiple copies of the strings in the file (easier to search, no chance of typos, etc.)